MudForm.IsTouched not updated when editing inside nested MudDataGrid

### Things to check

- [X] I have searched the **existing issues** for this bug
- [X] To rule out a caching problem I made sure the bug also happens in an **incognito tab**

### Bug type

Component

### Component name

MudForm and/or MudDataGrid

### What happened?

I have a MudForm which contains a MudDataGrid. The MudDataGrid has columns of type TemplateColumn for editing properties of a collection of items.

See here for an example: https://try.mudblazor.com/snippet/GYcoOjwSyvkBVpNn

When typing into the first input field ("test"), IsTouched changes to true.
But when typing into one of the 3 items, IsTouched is not affected, no MudForm.IsTouchedChanged event is raised.

### Expected behavior

The MudForm.IsTouched features should also work with components nested inside a child MudDataGrid.

### Reproduction link

https://try.mudblazor.com/snippet/GYcoOjwSyvkBVpNn

### Reproduction steps

1. Type into one of the items (1,2,3), IsTouched does not change (NOT OK).
2. Now type into the first text field ("test"), IsTouched changes to true (OK).


### Relevant log output

_No response_

### Version (bug)

7.8.0

### Version (working)

_No response_

### What browsers are you seeing the problem on?

Edge

### On which operating systems are you experiencing the issue?

Windows

### Pull Request

- [x] I would like to do a Pull Request

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Meanwhile I found a better workaround:
1. Create a class (e.g. `MyFormDataGrid`) that derives from `MudDataGrid`, has a `[CascadingParameter]` of type `IForm` and overrides `OnInitialized`.
2. In `OnInitialized` replace `MudDataGrid.Validator` with the cascading `IForm` value.
3. Use this class instead of `MudDataGrid` when an outer `MudForm` should be the "validation host".

```
public class MyFormDataGrid<T> : MudDataGrid<T>
{
    [CascadingParameter]
    public IForm? Form { get; set; }

    protected override void OnInitialized()
    {
        base.OnInitialized();

        if (this.Form is not null)
        {
            this.Validator = this.Form;
        }
    }
}

```

This works pretty well. No more `@bind-Value:after="this.Validate"` re-validation needed, plus the nested component validation states contribute to the outer `MudForm.IsValid` state as expected.

The only remaining problem is that now you have to explicitly specify the `T` parameter on each `<TemplateColumn>` element (plus `TProperty` for each `<PropertyColumn>`) to avoid strange Razor compilation errors like this:

```
...\obj\Debug\net8.0\Microsoft.CodeAnalysis.Razor.Compiler\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\Pages_MyPage_razor.g.cs(902,134,902,156): error CS0411: The type arguments for method 'TypeInference.CreatePropertyColumn_1<T, TProperty>(RenderTreeBuilder, int, int, Expression<Func<T, TProperty>>, int, string, int, string, int, string)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
```

As a workaround, replace markup like this:

```
<PropertyColumn Property="c => c.MyProp" />
```

with this:

```
<PropertyColumn T="Model" TProperty="string" Property="c => c.MyProp" />
```

(Just an example for items of type `Model` and `Model.MyProp` of type `string`.)

--- a/src/MudBlazor.Docs/Pages/Components/DataGrid/DataGridPage.razor
+++ b/src/MudBlazor.Docs/Pages/Components/DataGrid/DataGridPage.razor
@@ -346,5 +346,18 @@
                 <DataGridColumnsPanelExample />
             </SectionContent>
         </DocsPageSection>
+
+    <DocsPageSection>
+            <SectionHeader Title="Validator">
+                <Description>
+                    When a <CodeInline>&lt;MudDataGrid></CodeInline> is used as part of a <CodeInline>MudForm</CodeInline> you might want to
+                    bind the <CodeInline>Validator</CodeInline> property to the outer <CodeInline>MudForm</CodeInline> so that validation
+                    and <CodeInline>IsTouched</CodeInline> states are reflected in/propagated to the outer <CodeInline>MudForm</CodeInline>.
+                </Description>
+            </SectionHeader>
+            <SectionContent DarkenBackground="true" Code="@nameof(DataGridValidatorExample)" ShowCode="false" Block="true" FullWidth="true">
+                <DataGridValidatorExample />
+            </SectionContent>
+        </DocsPageSection>
     </DocsPageContent>
 </DocsPage>

--- a/src/MudBlazor.Docs/Pages/Components/DataGrid/DataGridPage.razor
+++ b/src/MudBlazor.Docs/Pages/Components/DataGrid/DataGridPage.razor
@@ -347,7 +347,7 @@
             </SectionContent>
         </DocsPageSection>
 
-    <DocsPageSection>
+        <DocsPageSection>
             <SectionHeader Title="Validator">
                 <Description>
                     When a <CodeInline>&lt;MudDataGrid></CodeInline> is used as part of a <CodeInline>MudForm</CodeInline> you might want to

--- a/src/MudBlazor.Docs/Pages/Components/DataGrid/Examples/DataGridValidatorExample.razor
+++ b/src/MudBlazor.Docs/Pages/Components/DataGrid/Examples/DataGridValidatorExample.razor
@@ -1,0 +1,68 @@
+﻿@namespace MudBlazor.Docs.Examples
+
+<MudForm @ref="_form"
+         IsTouchedChanged="this.StateHasChanged"
+         IsValidChanged="this.StateHasChanged">
+    <MudDataGrid T="Item"
+                 Items="_items"
+                 Validator="@(_bindValidatorToForm ? _form : _defaultValidator)">
+        <Columns>
+            <TemplateColumn Title="Name">
+                <CellTemplate>
+                    <MudTextField @bind-Value="context.Item.Name"
+                                  Immediate="true"
+                                  Validation="(string name) => ValidateName(name)" />
+                </CellTemplate>
+            </TemplateColumn>
+        </Columns>
+    </MudDataGrid>
+</MudForm>
+
+<MudStack Row>
+    <MudSwitch Color="Color.Primary" @bind-Value="_bindValidatorToForm">Bind MudDataGrid.Validator to MudForm?</MudSwitch>
+    <MudSwitch ReadOnly="true" Value="_form?.IsTouched">MudForm.IsTouched</MudSwitch>
+    <MudSwitch ReadOnly="true" Value="_form?.IsValid">MudForm.IsValid</MudSwitch>
+</MudStack>
+
+@code
+{
+    private readonly Interfaces.IForm _defaultValidator = new DataGridRowValidator();
+    private MudForm _form;
+    private bool _bindValidatorToForm = true;
+
+    private Item[] _items =
+    [
+        new() { Name = "valid" },
+        new() { Name = "invalid" }
+    ];
+
+    private static string ValidateName(string name)
+    {
+        if (string.IsNullOrWhiteSpace(name))
+        {
+            return "Name is required";
+        }
+
+        if (name == "invalid")
+        {
+            return "Name must not be 'invalid'";
+        }
+
+        return null;
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        await base.OnParametersSetAsync();
+        if (firstRender)
+        {
+            await _form.Validate();
+            _form.ResetTouched();
+        }
+    }
+
+    public class Item
+    {
+        public string Name { get; set; }
+    }
+}

--- a/src/MudBlazor.Docs/Pages/Components/DataGrid/Examples/DataGridValidatorExample.razor
+++ b/src/MudBlazor.Docs/Pages/Components/DataGrid/Examples/DataGridValidatorExample.razor
@@ -1,8 +1,6 @@
 ﻿@namespace MudBlazor.Docs.Examples
 
-<MudForm @ref="_form"
-         IsTouchedChanged="this.StateHasChanged"
-         IsValidChanged="this.StateHasChanged">
+<MudForm @ref="_form" ValidationDelay="0">
     <MudDataGrid T="Item"
                  Items="_items"
                  Validator="@(_bindValidatorToForm ? _form : _defaultValidator)">

--- a/src/MudBlazor.UnitTests.Viewer/TestComponents/DataGrid/DataGridValidatorTest.razor
+++ b/src/MudBlazor.UnitTests.Viewer/TestComponents/DataGrid/DataGridValidatorTest.razor
@@ -1,0 +1,29 @@
+﻿@namespace MudBlazor.UnitTests.TestComponents
+
+<MudForm @ref="_form" ValidationDelay="0">
+    <MudDataGrid Items="@_items" Validator="_form">
+        <Columns>
+            <TemplateColumn>
+                <CellTemplate>
+                    <MudTextField @bind-Value="context.Item.Name" Immediate="true" Required />
+                </CellTemplate>
+            </TemplateColumn>
+        </Columns>
+    </MudDataGrid>
+</MudForm>"
+
+@code {
+    public static string __description__ = "Changing the text field to a non-empty string should set form IsTouched=true and IsValid=true";
+
+    private MudForm _form;
+
+    private readonly Item[] _items =
+    [
+        new()
+    ];
+
+    public class Item
+    {
+        public string Name { get; set; }
+    }
+}

--- a/src/MudBlazor.UnitTests/Components/DataGridTests.cs
+++ b/src/MudBlazor.UnitTests/Components/DataGridTests.cs
@@ -4645,5 +4645,24 @@ namespace MudBlazor.UnitTests.Components
             items[2].TextContent.Should().Be("Paid training");
             items[3].TextContent.Should().Be("Untranslated");
         }
+
+        [Test]
+        public void DataGridValidatorFormBinding()
+        {
+            var comp = Context.RenderComponent<DataGridValidatorTest>();
+            var form = comp.FindComponent<MudForm>().Instance;
+            var dataGrid = comp.FindComponent<MudDataGrid<DataGridValidatorTest.Item>>().Instance;
+            dataGrid.Validator.Should().BeSameAs(form);
+
+            var textField = comp.FindComponent<MudTextField<string>>();
+            form.IsTouched.Should().BeFalse();
+            form.IsValid.Should().BeFalse();
+
+            // input into text field
+            textField.Find("input").Change("not empty");
+
+            form.IsTouched.Should().BeTrue();
+            form.IsValid.Should().BeTrue();
+        }
     }
 }

--- a/src/MudBlazor.UnitTests/Components/DataGridTests.cs
+++ b/src/MudBlazor.UnitTests/Components/DataGridTests.cs
@@ -4658,11 +4658,17 @@ namespace MudBlazor.UnitTests.Components
             form.IsTouched.Should().BeFalse();
             form.IsValid.Should().BeFalse();
 
-            // input into text field
-            textField.Find("input").Change("not empty");
+            // input valid value into text field
+            textField.Find("input").Input("not empty");
 
             form.IsTouched.Should().BeTrue();
             form.IsValid.Should().BeTrue();
+
+            // input invalid value into text field
+            textField.Find("input").Input("");
+
+            form.IsTouched.Should().BeTrue();
+            form.IsValid.Should().BeFalse();
         }
     }
 }

--- a/src/MudBlazor/Components/DataGrid/MudDataGrid.razor.cs
+++ b/src/MudBlazor/Components/DataGrid/MudDataGrid.razor.cs
@@ -1169,6 +1169,10 @@ namespace MudBlazor
         /// <summary>
         /// The validator which validates values in each row.
         /// </summary>
+        /// <remarks>
+        /// Use to bind validation and touched handling to outer form.
+        /// </remarks>
+        [Parameter]
         public Interfaces.IForm Validator { get; set; } = new DataGridRowValidator();
 
         internal Column<T> GroupedColumn

